<?xml version="1.0" encoding="utf-8" standalone="yes" ?>
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
  <channel>
    <title>Gerrit on Ganesh Kadam&#39;s Blog</title>
    <link>https://meganeshkadam.github.io/tags/gerrit/</link>
    <description>Recent content in Gerrit on Ganesh Kadam&#39;s Blog</description>
    <generator>Hugo -- gohugo.io</generator>
    <language>en-us</language>
    <lastBuildDate>Tue, 26 Apr 2016 01:00:12 +0530</lastBuildDate>
    <atom:link href="https://meganeshkadam.github.io/tags/gerrit/index.xml" rel="self" type="application/rss+xml" />
    
    <item>
      <title>Gerrit Isue</title>
      <link>https://meganeshkadam.github.io/post/gerrit/</link>
      <pubDate>Tue, 26 Apr 2016 01:00:12 +0530</pubDate>
      
      <guid>https://meganeshkadam.github.io/post/gerrit/</guid>
      <description>&lt;p&gt;I was working on shell a script. Now that it was complete I need to send that script for review.So I started to work on this. Unfortunately I got the following error which took me almost took a day to resolve on my side. I decided not to ask my mentors until and unless I am not able to rsolve this issue. This has been common issue amongst all of us interns!&lt;/p&gt;

&lt;pre&gt;&lt;code&gt;[gkadam@dhcp201-114 cert-openstack-scripts]$ git review -R
Could not connect to gerrit.
Enter your gerrit username: gkadam@redhat.com
Trying again with ssh://gkadam@redhat.com@code.engineering.redhat.com:22/cert-openstack-scripts
&amp;lt;traceback object at 0x7fa7a448cb00&amp;gt;
We don&#39;t know where your gerrit is. Please manually create a remote
named &amp;quot;gerrit&amp;quot; and try again.
Could not connect to gerrit at ssh://gkadam@redhat.com@code.engineering.redhat.com:22/cert-openstack-acripts
Traceback (most recent call last):
File &amp;quot;/usr/bin/git-review&amp;quot;, line 11, in &amp;lt;module&amp;gt;
sys.exit(main())
File &amp;quot;/usr/lib/python2.7/site-packages/git_review/cmd.py&amp;quot;, line 1534, in main
sys.exit(e.EXIT_CODE)
AttributeError: &#39;GitReviewException&#39; object has no attribute &#39;EXIT_CODE&#39;
&lt;/code&gt;&lt;/pre&gt;

&lt;p&gt;After going through lots of google serarches I realized that, the things that were going wrong are already mentioned in the error code that I was getting. Before this I made a very silly mistake of deleting the existing file in that directory from which I have to send the file for review. So just after knowing this issue, I copied the new script in to the same directory again. Then I renamed to it&amp;rsquo;s earlier name and performed following steps :&lt;/p&gt;

&lt;ol&gt;
&lt;li&gt;Cloned the repository to new location :&lt;/li&gt;
&lt;/ol&gt;

&lt;!-- --&gt;

&lt;pre&gt;&lt;code&gt;[gkadam@dhcp201-114 ~]$ git clone ssh://gkadam@code.engineering.redhat.com:22/cert-openstack-scripts
Cloning into &#39;cert-openstack-scripts&#39;...
Checking connectivity... done.
warning: remote HEAD refers to nonexistent ref, unable to checkout.
&lt;/code&gt;&lt;/pre&gt;

&lt;ol&gt;
&lt;li&gt;Added the remote repository as we can see in the above error!&lt;/li&gt;
&lt;/ol&gt;

&lt;!-- --&gt;

&lt;pre&gt;&lt;code&gt;[gkadam@dhcp201-114 cert-openstack-scripts]$ git remote add gerrit ssh://gkadam@code.engineering.redhat.com:22/cert-openstack-scripts
&lt;/code&gt;&lt;/pre&gt;

&lt;ol&gt;
&lt;li&gt;Then did a git review after adding the script and commiting it in that dierctory :&lt;/li&gt;
&lt;/ol&gt;

&lt;!-- --&gt;

&lt;pre&gt;&lt;code&gt;[gkadam@dhcp201-114 cert-openstack-scripts]$ git review -R
Your change was committed before the commit hook was installed.
Amending the commit to add a gerrit change id.
remote: Processing changes: new: 1, refs: 1, done.
remote: (W) 9dedc09: commit subject &amp;gt;65 characters; use shorter first paragraph
remote: (W) 9dedc09: too many commit message lines longer than 70 characters; manually wrap lines
remote:
remote: New Changes:
remote:https://code.engineering.redhat.com/gerrit/73014 Add user validation, and create functions : create_member,create_network and remote:
To ssh://gkadam@code.engineering.redhat.com:22/cert-openstack-scripts
* [new branch]      HEAD -&amp;gt; refs/publish/master
&lt;/code&gt;&lt;/pre&gt;

&lt;p&gt;This was the workaround for my problem that I was facing.&lt;/p&gt;

&lt;p&gt;Here is the link to the similar issue on git : &lt;a href=&#34;https://help.github.com/articles/error-remote-head-refers-to-nonexistent-ref-unable-to-checkout/&#34;&gt;Git remote HEAD Issue&lt;/a&gt;&lt;/p&gt;
</description>
    </item>
    
  </channel>
</rss>